spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesCatalogWatch.java

```diff
@@ -45,13 +47,17 @@ public class KubernetesCatalogWatch implements ApplicationEventPublisherAware {
 
 	private final KubernetesDiscoveryProperties properties;
 
+	private final KubernetesNamespaceProvider namespaceProvider;
+
 	private volatile List<EndpointNameAndNamespace> catalogEndpointsState = null;
 
 	private ApplicationEventPublisher publisher;
 
-	public KubernetesCatalogWatch(KubernetesClient kubernetesClient, KubernetesDiscoveryProperties properties) {
+	public KubernetesCatalogWatch(KubernetesClient kubernetesClient, KubernetesDiscoveryProperties properties,
```

Comment:
@ryanjbaxter need your wisdom here. If we do it like this - use the constructor to inject the `KubernetesNamespaceProvider` - this is a breaking change.

We can `@Autowire` the `Environment` and create the `KubernetesNamespaceProvider` from it - it would not be a breaking change, but a bit ugly (imho). So, which one would you favor?

Reply:
pretty please? God, I feel weird tagging you twice, I'm sorry upfront!